File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__vision_test.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int level = 4;
        int n = int.Parse( ReadLine() );
        int[] ng = new int[4];
        int[] ok = new int[4];
        char[] lvl = new char[] { 'A', 'B', 'C', 'D', 'E' };
        string[] exm = new string[] { "TA", "TB", "TC", "TD" };

        foreach( int _ in Range( 0, n ) ) {
            string[] tj = ReadLine().Split( ' ' );
            var c = System.StringComparison.Ordinal;

            foreach( int i in Range( 0, 4 ).Where( i => string.Compare( tj[0], exm[i], c ) == 0 ) ) {
                if( string.Compare( tj[1], "ng", c ) == 0 ) {
                    ng[i]++;
                } else if( string.Compare( tj[1], "ok", c ) == 0 ) {
                    ok[i]++;
                }
                if( 2 <= ok[i] && ng[i] < 2 ) {
                    level = System.Math.Min( level, i );
                }
            }
        }
        WriteLine( lvl[level] );
        return;
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__vision_test.py
# 実行時間 20ms
n = int(input())
l = 4
ng = [0, 0, 0, 0]
ok = [0, 0, 0, 0]
lv = ["A", "B", "C", "D", "E"]
ex = ["TA", "TB", "TC", "TD"]

for _ in range(n):
    t = input().split()

    for i in range(4):
        if t[0] == ex[i]:
            if t[1] == "ng":
                ng[i] += 1
            elif t[1] == "ok":
                ok[i] += 1
            if 2 <= ok[i] and ng[i] < 2:
                l = min(l, i)
print(lv[l])
